crucible-mir/src/Mir/Trans.hs

```
@@ -1364,8 +1364,23 @@ evalRval rv@(M.RAdtAg (M.AdtAg adt agv ops ty optField)) = do
             case adt^.adtkind of
                 M.Struct -> buildStruct adt es
                 M.Enum _ -> buildEnum adt (fromInteger agv) es
-                M.Union -> do
-                    mirFail $ "evalRval: Union types are unsupported, for " ++ show (adt ^. adtname)
+                -- This is a hack. crucible-mir doesn't _actually_ support union
+                -- types, and if you try simulating a code that performs an
+                -- operation on a union-typed value, then it will fail at
+                -- simulation time. Nevertheless, we intentionally choose to
+                -- give unions a dummy value here instead of outright failing so
+                -- that crucible-mir can translate some union-typed declarations
+                -- without crashing during MIR translation.
+                --
+                -- We apply `packAny` here to match the similar hack described
+                -- in the union case of `Mir.Trans.tyToRepr`, which gives
+                -- union-typed values a dummy representation of `AnyRepr`.
+                --
+                -- See #1429 for the larger issue of properly supporting union
+                -- types.
+                M.Union ->
+                  let dummyExpr = R.App $ usizeLit 0xdeadbeef
+                   in pure $ packAny $ MirExp knownRepr dummyExpr
```

My worry with this approach is that it allows some things that we can't support long-term, so when we implement unions properly with MirAggregate, we will get regressions.  For example:
```Rust
union U {
  x: i32,
  y: u8,
}
let u = if bool::symbolic("cond") {
  U { x: 1 }
} else {
  U { y: 2 }
};
// `u` is unused afterward
```
This would be accepted with this approach because both branches would produce `AnyRepr` wrapping `UsizeRepr`, and same-typed `AnyRepr`s can be muxed together (IIRC).  But the future `MirAggregate` version would produce `MirAggregateRepr`s wrapping `BVRepr 32` and `BVRepr 8`, which can't be muxed together.

I think the current `MirAggregate` implementation is already sufficient to implement unions properly.  But if you want to stick with this stopgap implementation, please change it to wrap the actual value instead of a dummy `UsizeRepr`, so that muxing fails in the same way that it would in the `MirAggregate` version.